tests/js/session-tests.js

```diff
@@ -542,12 +542,17 @@ module.exports = {
                     schema: [{ name: 'Dog', properties: { name: 'string' } }],
                 };
 
-                const realm = new Realm(config);
-                return new Promise((resolve, reject) => {
-                    realm.syncSession.addProgressNotification('download', 'reportIndefinitely', resolve);
-                    setTimeout(function() {
-                        reject("Progress Notifications API failed to call progress callback for Realm constructor");
-                    }, 5000);
+                return Realm.open(config).then((realm) => {
+                    return new Promise((resolve, reject) => {
+                        realm.syncSession.addProgressNotification('download', 'reportIndefinitely', (x, y) => {
+                            if (x === y) {
```

Comment:
x, y are not the most obvious... perhaps better names? @kneth

Reply:
True. I will change the names in a separate PR.